pkg/test/echo/client/client.go

```diff
@@ -52,7 +52,11 @@ func New(address string, tlsSettings *common.TLSSettings) (*Instance, error) {
 			return nil, err
 		}
 
-		certPool := x509.NewCertPool()
+		var certPool *x509.CertPool
+		if tlsSettings.RootCert == "" {
+			certPool, _ = x509.SystemCertPool()
```

Comment:
This still doesn't match your comment in model.go and has problems if RootCert != "" (certPool would be nil).

To implement as the comment you write in model.go, you should always call SystemCertPool() to get the certPool, and append tlsSettings.RootCert if it's not empty.